File: Siav/SiavWeb/src/main/java/co/com/siav/rest/services/ProveedorServices.java
package co.com.siav.rest.services;

import java.util.List;

import javax.enterprise.context.RequestScoped;
import javax.inject.Inject;
import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import co.com.siav.bean.ProveedorBean;
import co.com.siav.entities.Proveedor;
import co.com.siav.request.ProveedorRequest;
import co.com.siav.response.MensajeResponse;

@RequestScoped
@Path("proveedor/")
public class ProveedorServices {
	
	@Inject
	private ProveedorBean bean;
	
	@POST
	@Path("crear")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	public MensajeResponse crear(ProveedorRequest request){
		return bean.crear(request);
	}
	
	@POST
	@Path("actualizar")
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	public MensajeResponse actualizar(ProveedorRequest request){
		return bean.actualizar(request);
	}
	
	@GET
	@Path("consultar")
	@Produces(MediaType.APPLICATION_JSON)
	public List<Proveedor> consultar(){
		return bean.consultar();
	}
	
	@GET
	@Path("consultar/nombres")
	@Produces(MediaType.APPLICATION_JSON)
	public List<String> getNombres(){
		return bean.consultarNombres();
	}
	
}

File: Siav/SiavWeb/src/main/webapp/scripts/services/proveedor-services.js
/*global define*/
'use strict';

define(['siav-module', 'constantes'], function (app) {
    app.service('proveedorServices', ['$http', '$q', 'CONSTANTES', function ($http, $q, CONSTANTES) {
    	 var contrato = {
    			 consultar : consultar,
    			 consultarNombres : consultarNombres,
    			 crear : crear,
    			 actualizar : actualizar
    	 };
    	 
    	 function crear(proveedor){
    		 var request = $http.post(CONSTANTES.SRV.PROVEEDOR_CONSULTAR_CREAR, proveedor);
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
    	 
    	 function actualizar(proveedor){
    		 var request = $http.post(CONSTANTES.SRV.PROVEEDOR_CONSULTAR_ACTUALIZAR, proveedor);
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
    	 
    	 function consultar(){
    		 var request = $http.get(CONSTANTES.SRV.PROVEEDOR_CONSULTAR, {isArray : true});
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
    	 
    	 function consultarNombres(){
    		 var request = $http.get(CONSTANTES.SRV.PROVEEDOR_CONSULTAR_NOMBRES, {isArray : true});
    		 return (request.then(function(response) {
 	    		return response.data;
	    	}));
    	 }
   	 
    	 return contrato;
    }]);
});